### PR TITLE
Remove partition key and add and query support

### PR DESCRIPTION
**Pull request by sophiaso**

### CODE REVIEW

File: src/main/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplate.java [hunk 1]
```diff
@@ -170,15 +166,10 @@ public void setApplicationContext(ApplicationContext applicationContext) throws
...
-        if (partitionKeyFieldName != null && !partitionKeyFieldName.isEmpty()) {
-            sqlQuerySpec = new SqlQuerySpec("SELECT * FROM root c WHERE c." + partitionKeyFieldName + "=@partition",
-                    new SqlParameterCollection(new SqlParameter("@partition", partitionKeyFieldValue)));
-            feedOptions.setPartitionKey(new PartitionKey(partitionKeyFieldValue));
-        }
+        final SqlQuerySpec sqlQuerySpec = new SqlQuerySpec("SELECT * FROM root c");
 
         final List<Document> results = documentDbFactory.getDocumentClient()
-                .queryDocuments(collections.get(0).getSelfLink(), sqlQuerySpec, feedOptions, partitionKeyFieldName)
+                .queryDocuments(collections.get(0).getSelfLink(), sqlQuerySpec, feedOptions)
```
Comment (yungez) [inline src/main/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplate.java]: set feedOption always no matter collection partitioned or not?

File: src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java [hunk 2]
```diff
@@ -85,50 +88,66 @@ public void cleanup() {
...
-        final Person nullResult = dbTemplate.findById(Person.class.getSimpleName(),
-                TEST_NOTEXIST_ID, Person.class, TEST_PERSON.getLastName());
-        assertThat(nullResult).isNull();
+        final List<Person> result = dbTemplate.find(query, Person.class, Person.class.getSimpleName());
+        assertThat(result.size()).isEqualTo(1);
+        assertTrue(result.get(0).equals(TEST_PERSON));
+    }
+
+    @Test
+    public void testFindByNonExistIdWithPartition() {
```
Comment (yungez) [inline src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java]: might need to consider move partition related tests into seperate test class.

File: src/main/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplate.java (hunk 1 above)
Comment (sophiaso) [reply src/main/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplate.java]: Thanks, for the findAll, the enableCrossRegion will be enabled as no PartitionKey value is provided, but I have made change for other 2 methods in this [commit](https://github.com/Microsoft/spring-data-documentdb/pull/37/commits/25b4a2eaf311ea23f868ede9b86b41484a25dab8)

File: src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java (hunk 2 above)
Comment (sophiaso) [reply src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java]: Split the test into 2 test classes [here](https://github.com/Microsoft/spring-data-documentdb/pull/37/commits/de7076b777bcc8ef318ef02d30a444fc7d332614).

File: pom.xml [hunk 3]
```diff
@@ -132,6 +132,12 @@
             <artifactId>lombok</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
```
Comment (yungez) [inline pom.xml]: what's this package used for?

File: QueryPartitionedCollection.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+### How to Query Partitioned CosmosDB Collection
```
Comment (yungez) [inline QueryPartitionedCollection.md]: DocumentDB

File: QueryPartitionedCollection.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    public class Address {
+        @Id
+        String postalCode;
+        String street;
+        @PartitionKey
+        String city;
+    }
+```
+
+How to write the repository interface:
```
Comment (yungez) [inline QueryPartitionedCollection.md]: better to have some description:  Add custom method with `And` operator to query..

File: QueryPartitionedCollection.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+```
+    @Repository
+    public interface AddressRepository extends DocumentDbRepository<Address, String> {
+        void deleteByPostalCodeAndCity(String postalCode, String city);
+        void deleteByCity(String city);   
+        List<Address> findByPostalCodeAndCity(String postalCode, String city);
```
Comment (yungez) [inline QueryPartitionedCollection.md]: add comments to explain what's the method used for: when query postalcode, need specify 'andcity'

**Pull request merged**